kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -109,38 +60,48 @@ public static String consumeMessageWithTestClients(String deployNamespace, Strin
                 .until(() -> {
```

Comment:
what you have here is ugly.  You've repeated the computation getting the log so you can return it.  Awaitability offers a nicer solution - there's a form of until that can return the data.

```
        return await().alias("Consumer waiting to receive messages")
                .ignoreException(KubernetesClientException.class)
                .atMost(timeout)
                .until(() -> {
                    if (kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName).get() != null) {
                        return kubeClient().logsInSpecificNamespace(deployNamespace, podName);
                    }
                    return null;
                }, m -> m != null && m.contains(messageToSeek));
```